tests/features/mocking.py

```diff
@@ -0,0 +1,1404 @@
+"""Tests for flexmock mocking feature."""
+# pylint: disable=missing-docstring,no-self-use,no-member,too-many-lines
+import os
+import random
+import re
+import sys
+
+from flexmock import exceptions, flexmock
+from flexmock._api import AT_LEAST, AT_MOST, EXACTLY, FlexmockContainer, Mock, flexmock_teardown
+from tests import some_module
+from tests.utils import assert_raises
+
+
+class MockingTestCase:
```

Comment:
That is one big class we might want to split, though maybe another time

Reply:
Yeah, there are some test cases that we could move to smaller modules. I'll leave that for later.